test/e2e/azure_test.go

```diff
@@ -597,4 +597,55 @@ var _ = Describe("Workload cluster creation", func() {
 			})
 		})
 	})
+
+	Context("Creating clusters using clusterclass [REQUIRED]", func() {
+		It("with a single control plane node, one linux worker node, and one windows worker node", func() {
+			// use "cc" as spec name because natgw pip name exceeds limit.
+			clusterName = getClusterName(clusterNamePrefix, "cc")
+
+			// Opt into using windows with prow template
+			Expect(os.Setenv("WINDOWS_WORKER_MACHINE_COUNT", "1")).To(Succeed())
+			Expect(os.Setenv("K8S_FEATURE_GATES", "WindowsHostProcessContainers=true")).To(Succeed())
+
+			// Get cluster class yaml. This is a hacky way to get cluster class with existing clusterctl api.
```

Comment:
what would it take to make this not hacky?

Reply:
Ideally we want something in capi test library like `capi.CreateTopology` that applies kustomization and applies the topology manifest. Functionally it would be the same, just a better api. Also having a `topology` flavor seems kind of misleading since we are not creating any cluster with `topology` flavor.

Reply:
is there a CAPI issue for this? If not, let's create one. Seems like there would be a great benefit in sharing this with other providers.

> Also having a topology flavor seems kind of misleading since we are not creating any cluster with topology flavor.

agreed